### PR TITLE
[create-dmg/xxx] conan v2 migration

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **create-dmg/xxx**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/create-dmg/all/conanfile.py [hunk 1]
```diff
@@ -13,34 +13,39 @@ class CreateDmgConan(ConanFile):
     topics = "command-line", "dmg"
     homepage = "https://github.com/create-dmg/create-dmg"
     url = "https://github.com/conan-io/conan-center-index"
-    settings = "os", "arch", "compiler", "build_type"
+    settings = "os"
```
Comment (prince-chrismc) [inline recipes/create-dmg/all/conanfile.py:16]: Sicne this is a tool, it should have all the settings here but delete them from package id

```suggestion
    settings = "os", "arch", "compiler", "build_type"
```

File: recipes/create-dmg/all/conanfile.py [hunk 2]
```diff
@@ -13,34 +13,39 @@ class CreateDmgConan(ConanFile):
...
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, pattern="create-dmg", dst=os.path.join(self.package_folder, "bin"), src=self.source_folder)
+        copy(self, pattern="*", dst=os.path.join(self.package_folder, "res", "create-dmg", "support"), src=os.path.join(self.source_folder,"support"))
 
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
```
Comment (prince-chrismc) [inline recipes/create-dmg/all/conanfile.py:42]:
```suggestion
        del self.settings.compiler
        del self.settings.build_type
```

File: recipes/create-dmg/all/conanfile.py [hunk 3]
```diff
@@ -13,34 +13,39 @@ class CreateDmgConan(ConanFile):
...
     url = "https://github.com/conan-io/conan-center-index"
-    settings = "os", "arch", "compiler", "build_type"
+    settings = "os"
     exports_sources = 'patches/**'
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def layout(self):
+        pass
```
Comment (prince-chrismc) [inline recipes/create-dmg/all/conanfile.py:20]: We always should have a layout for 2.0 :)

```suggestion
        basic_layout(self)
```

**Review by prince-chrismc — COMMENTED** [review]
two small comments

**Comment by MartinDelille** [comment]
@prince-chrismc I applied your suggestion. Regarding the layout, I made the change according to this example: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/prebuilt_tool_package/all/conanfile.py#L21-L23

Shall the example be fixed ?

**Comment by prince-chrismc** [comment]
> Regarding the layout, I made the change according to this example: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/prebuilt_tool_package/all/conanfile.py#L21-L23
> 
> Shall the example be fixed ?

Nop you are absolutely correct, this is a tool and you setup the cpp_info dirs correctly so this should be good!

**Review by ericLemanissier — APPROVED** [review]

**Comment by SpaceIm** [comment]
I've submitted https://github.com/conan-io/conan-center-index/pull/14119 because there was no reason to download source code in build(), nor to not clear package id completely.

**Pull request merged**